### ISSUE TITLE
Problem when Submitting a TaxBrain Job

### ISSUE THREAD

**Issue by brittainhard** [open-source-economics/Tax-Calculator#1437]
Getting this as a traceback on 0.9.0. Looks like another pandas issue. @jdebacker @PeterDSteinberg 
```
[2017-06-16 09:30:12,575: ERROR/MainProcess] Task celery_tasks.dropq_task_async[0dc45265-c546-42fb-b89a-f410f2ece339] raised unexpected: IndexError('positional indexers are out-of-bounds',)
Traceback (most recent call last):
  File "/Users/bhard/anaconda/envs/aei_dropq/lib/python2.7/site-packages/celery/app/trace.py", line 367, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/Users/bhard/anaconda/envs/aei_dropq/lib/python2.7/site-packages/celery/app/trace.py", line 622, in __protected_call__
    return self.run(*args, **kwargs)
  File "/Users/bhard/Documents/webapp-public/deploy/taxbrain_server/celery_tasks.py", line 116, in dropq_task_async
    return dropq_task(year, user_mods, first_budget_year, beh_params, tax_dta_full)
  File "/Users/bhard/Documents/webapp-public/deploy/taxbrain_server/celery_tasks.py", line 93, in dropq_task
    fiscal_tot_i_bl, fiscal_tot_i_ref) = taxcalc.dropq.run_nth_year_tax_calc_model(**kw)
  File "/Users/bhard/anaconda/envs/aei_dropq/lib/python2.7/site-packages/taxcalc/dropq/dropq.py", line 54, in run_nth_year_tax_calc_model
    mask_computed=True)
  File "/Users/bhard/anaconda/envs/aei_dropq/lib/python2.7/site-packages/taxcalc/dropq/dropq_utils.py", line 80, in dropq_calculate
    gfactors=growfactors_pre)
  File "/Users/bhard/anaconda/envs/aei_dropq/lib/python2.7/site-packages/taxcalc/records.py", line 150, in __init__
    self.WT = self.WT.iloc[self.index]
  File "/Users/bhard/anaconda/envs/aei_dropq/lib/python2.7/site-packages/pandas/core/indexing.py", line 1328, in __getitem__
    return self._getitem_axis(key, axis=0)
  File "/Users/bhard/anaconda/envs/aei_dropq/lib/python2.7/site-packages/pandas/core/indexing.py", line 1738, in _getitem_axis
    return self._get_list_axis(key, axis=axis)
  File "/Users/bhard/anaconda/envs/aei_dropq/lib/python2.7/site-packages/pandas/core/indexing.py", line 1718, in _get_list_axis
    raise IndexError("positional indexers are out-of-bounds")
IndexError: positional indexers are out-of-bounds
```

**Comment by MattHJensen** [comment]
@brittainhard, have you been able to replicate this with taxcalc directly? It would facilitate help from tax-calculator developers if you show how to replicate w taxcalc.

**Comment by brittainhard** [comment]
@MattHJensen I dont use taxcalc directly, but i'll see if I can replicate it. @jdebacker do you have any opinions on this?

**Comment by brittainhard** [comment]
Was able to recreate this by running `taxcalc.dropq.run_nth_year_tax_calc_model(**data)` where `data` is the dict run through the function. I managed to save the offending data to a pickle file, but its about 500mb large. I can send this file if need be.

**Comment by MattHJensen** [comment]
Could you paste here:

a) the code that you used to generate `taxref_df` from the puf.csv file from this week. 
b) the other arguments (`year_n`, `start_year`, `user_mods`) 


`run_nth_year_tax_calc_model(year_n, start_year, taxrec_df, user_mods, return_json=True):`

**Comment by brittainhard** [comment]
As far as the DataFrame used, it was generated from running a task off of the webapp. I have a copy of that data saved in a pickle file. As for what `puf.csv` file I was using at the time I really can't tell. I'm doing all of this on a local deployment.

The other arguments are

```
year_n : 0, start_year: 2017, (df), user_mods:
{u'behavior': {},
 u'consumption': {},
 u'gdp_elasticity': {},
 u'growdiff_baseline': {},
 u'growdiff_response': {},
 u'policy': {2017: {u'_ALD_InvInc_ec_base_RyanBrady': [False],
   u'_CG_nodiff': [False],
   u'_CTC_new_refund_limited': [False],
   u'_EITC_indiv': [False],
   u'_ID_BenefitCap_Switch': [[True, True, True, True, 1.0, 1.0, 1.0]],
   u'_ID_BenefitSurtax_Switch': [[True, True, True, True, True, True, True]],
   u'_II_no_em_nu18': [False],
   u'_NIIT_PT_taxed': [False]}}} 
```

**Comment by MattHJensen** [comment]
@brittainhard, the new puf.csv file is a requirement for taxcalc 0.9.0. Could you see if you can replicate the error with that? It looks like taxpuf 1.2.0 contains the new puf.

**Comment by brittainhard** [comment]
Sure thing

**Comment by jdebacker** [comment]
@brittainhard I don't have specific ideas since I've not run stuff in the way you are.  I did run into a similar error when trying to run B-Tax without setting the environment to the Tax-Calculator environment.  I believe it is a pandas version issue.

Also, I've noticed using pandas and pickle on several Python versions recently (2.7, 3.4, 3.6) that there are often issues with reading objects created in one version with a different version (e.g., even consecutive versions of pickle can't read each others files).  My solution for pickles was:

```
TaxFunc_pkl = "TxFuncEst_baseline.pkl" with open(TaxFunc_pkl, 'rb') as f:
            dict_params = pickle.load(f, encoding='latin1')
```

For pandas, it has been to just pin to a different version.
